core/modules/filters/interleave.js

```diff
@@ -0,0 +1,75 @@
+/*\
+title: $:/core/modules/filters/interleave.js
+type: application/javascript
+module-type: filteroperator
+
+Interleave two or more lists one item at a time
+
+\*/
+(function(){
+
+	/*jslint node: true, browser: true */
+	/*global $tw: false */
+	"use strict";
+
+	/*
+	Interleave two or more lists
+	*/
+	exports.interleave = function(source,operator,options) {
+		var allowDuplicates = false;
+		switch(operator.suffix) {
+			case "raw":
+				allowDuplicates = true;
+				break;
+			case "dedupe":
+				allowDuplicates = false;
+				break;
+		}
+		var results = new $tw.utils.LinkedList();
+		var pushResult = allowDuplicates ? results.push.bind(results) : results.pushTop.bind(results);
+
+		// Alternately, could use the following function definition instead:
+		// function pushResult(item) {
+		// 	if(allowDuplicates) {
+		// 		results.push(item);
+		// 	} else {
+		// 		results.pushTop(item);
+		// 	}
+		// }
+
```

Comment:
Perhaps it's my functional-programming background, but I like the elegance of choosing the function once and not having to do the `if(allowDuplicates)` check every time. OTOH, any decent processor will quickly [branch-optimize](https://stackoverflow.com/a/11227902) that `if` condition since it's going to follow exactly the same branch every time it's called (within a single invocation of the filter, at least), so there won't really be any noticeable performance loss from going through the `if(allowDuplicates)` instruction every time.

Granted, in Javascript assigning an object method to a variable to call it later is a bit less elegant, because you have to write `foo.method.bind(foo)` rather than assigning `var myFunc = foo.method` the way you could in, say, Python. But I still find that one-liner to be quite clear, and actually *easier* to read and understand (for me) than the spelled-out function version.